tokio-util/src/codec/framed.rs

```diff
@@ -123,10 +95,17 @@ where
     /// [`split`]: https://docs.rs/futures/0.3/futures/stream/trait.StreamExt.html#method.split
     pub fn with_capacity(inner: T, codec: U, capacity: usize) -> Framed<T, U> {
         Framed {
-            inner: framed_read2_with_buffer(
-                framed_write2(Fuse { io: inner, codec }),
-                BytesMut::with_capacity(capacity),
-            ),
+            inner: FramedImpl {
+                inner,
+                codec,
+                state: RWFrames {
+                    read: ReadFrame {
+                        buffer: BytesMut::with_capacity(capacity),
+                        ..ReadFrame::default()
+                    },
```

Comment:
Same deal here. I guess this could just do `ReadFrame::from(BytesMut::with_capacity(capacity))` if we fix the other one.